src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/AccessTokenFactory.php

```diff
@@ -69,19 +69,20 @@ public function getKey(): string
         return 'access_token';
     }
 
-    public function createAuthenticator(ContainerBuilder $container, string $firewallName, array $config, string $userProviderId): string
+    public function createAuthenticator(ContainerBuilder $container, string $firewallName, array $config, ?string $userProviderId): string
```

Comment:
This should be reverted, we always pass a user provider in here.

We can improve this, but let's do that DX improvement in 6.3 (to not make the scope of this change unnecessary bigger).

Reply:
Alright, it was mainly a guess anyway, as I don't have the complete picture of how this factory works combined with the configs yet, so any suggestion like this is appreciated.

Reply:
This method is called by the [`SecurityExtension`](https://github.com/symfony/symfony/blob/521d210ca16470437e5c5143c80ab1fb3dc40820/src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php#L614), which receives the configuration and configures the container (services/parameters/etc).

In there, we currently [require configuring a user provider](https://github.com/symfony/symfony/blob/521d210ca16470437e5c5143c80ab1fb3dc40820/src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php#L640-L677) - or default to the configured user providers if there is only one. In the 6.3, we might want to take a look at allowing the user provider to be `null` on stateless firewalls, to avoid having to configure an empty in-memory user provider.
(statefull firewalls still require a user provider, as they need to be able to refresh the user on each subsequent request)